File: front/src/constant.js
export default {
  code: {
    error: -1,
    success: 200,
    sys_err: 500,
    not_found: 404,
    account_exist: 40301,
  },

  identity: {
    student: "STUDENT",
    teacher: "TEACHER",
  },

  sex: {
    female: "F",
    male: "M",
  },

  paper_state: {
    preparing: "PREPARING",
    waiting: "WAITING",
    starting: "STARTING",
    correcting: "CORRECTING",
    end: "END",
  },

  question_type: {
    choose: "CHO",
    fill: "FIL",
    subject: "SUB",
  },

  exam_paper_state: {
    ongoing: "ONGOING",
    cheating: "CHEATING",
    // 批改中
    correcting: "CORRECTING",
    finished: "FINISHED",
  },
  signal: {
    SIGNAL_TYPE_JOIN: "join",
    SIGNAL_TYPE_RESP_JOIN: "resp-join",
    SIGNAL_TYPE_LEAVE: "leave",
    SIGNAL_TYPE_NEW_PEER: "new-peer",
    SIGNAL_TYPE_PEER_LEAVE: "peer-leave",
    SIGNAL_TYPE_OFFER: "offer",
    SIGNAL_TYPE_ANSWER: "answer",
    SIGNAL_TYPE_CANDIDATE: "candidate",
  },
};

File: src/main/java/com/online_exam_sys/util/Constant.java
package com.online_exam_sys.util;

public interface Constant {
    interface code {
        int error = -1;
        int success = 200;
        int sys_err = 500;
        int not_found = 404;
        int account_exist = 40301;
    }

    interface sex {
        String female = "F";
        String male = "M";
    }

    interface paper_state {
        String preparing = "PREPARING";
        String waiting = "WAITING";
        String starting = "STARTING";
        String end = "END";
    }

    interface question_type {
        String choose = "CHO";
        String fill = "FIL";
        String subject = "SUB";
    }

    interface exam_paper_state {
        String ongoing = "ONGOING";
        String cheating = "CHEATING";
        // 批改中
        String correcting = "CORRECTING";
        String finished = "FINISHED";
    }

    interface signal {
        String SIGNAL_TYPE_JOIN = "join";
        String SIGNAL_TYPE_RESP_JOIN = "resp-join";
        String SIGNAL_TYPE_LEAVE = "leave";
        String SIGNAL_TYPE_NEW_PEER = "new-peer";
        String SIGNAL_TYPE_PEER_LEAVE = "peer-leave";
        String SIGNAL_TYPE_OFFER = "offer";
        String SIGNAL_TYPE_ANSWER = "answer";
        String SIGNAL_TYPE_CANDIDATE = "candidate";
    }
}
